flow/environments/xsede.py

```diff
@@ -215,8 +215,69 @@ def add_args(cls, parser):
         )
 
 
+class ExpanseEnvironment(DefaultSlurmEnvironment):
+    """Environment profile for the Expanse supercomputer.
+
+    https://www.sdsc.edu/support/user_guides/expanse.html
+    """
+
+    hostname_pattern = r".*\.expanse\.sdsc\.edu$"
+    template = "expanse.sh"
+    cores_per_node = 128
+    gpus_per_node = 4
+
+    @classmethod
+    def _get_mpi_prefix(cls, operation):
+        if operation.directives["nranks"] == 0:
+            return ""
+        # We must specify the pmi2 option as the SLURM configuration of Expanse uses the
+        # special pmi2 preprocessing. See `man srun` on an Expanse login node.
+        base_prefix = "srun --mpi=pmi2"
```

Comment:
This will work for codes built on expanse with their MPI library, but it does not work with the `glotzerlab-software` containers. Those need to be launched with `mpirun`. I have tried but not succeeded in the past to get `srun` to pass MPI configurations through to singularity.

I haven't tested if codes built on the system also work with `mpirun` or not.